.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: build
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [stable, nightly]
+
+    steps:
+    - uses: hecrj/setup-rust-action@v1
+      with:
+        rust-version: ${{ matrix.rust }}
+        components: rustfmt, clippy
```

Comment:
please remove the components here and move clippy to a separate job as well

Reply:
Yep. Wondering if a separate check task is necessary as clippy does run check (https://stackoverflow.com/questions/57449356/is-cargo-clippy-a-superset-of-cargo-check)